### PR TITLE
Bump VisualStudioVersion to 16.0

### PR DESCRIPTION
**Pull request by rainersigwald**
This will allow MSBuild to find Visual Studio 16 instances and is part
of the larger versioning approach for MSBuild 16.

### CODE REVIEW

File: src/Shared/Constants.cs [hunk 1]
```diff
@@ -58,12 +58,12 @@ internal static class MSBuildConstants
...
         /// The most current Visual Studio Version known to this version of MSBuild.
         /// </summary>
-        internal const string CurrentVisualStudioVersion = "15.0";
+        internal const string CurrentVisualStudioVersion = "16.0";
 
         /// <summary>
         /// The most current ToolsVersion known to this version of MSBuild.
         /// </summary>
-        internal const string CurrentToolsVersion = CurrentVisualStudioVersion;
+        internal const string CurrentToolsVersion = "15.0";
```
Comment (rainersigwald) [inline src/Shared/Constants.cs]: This was the line I had missed that caused it to fail a bajillion tests. The fix was . . . not hard.

Comment (cdmihai) [reply src/Shared/Constants.cs]: Why is it 15 and not 16?

Comment (cdmihai) [reply src/Shared/Constants.cs]: Oh, toolsversion, nvm :)

File: src/Shared/Constants.cs [hunk 2]
```diff
@@ -82,13 +82,7 @@ internal static class MSBuildConstants
...
         /// Current version of this MSBuild Engine assembly in the form, e.g, "12.0"
         /// </summary>
-        internal static string CurrentProductVersion
-        {
-            get
-            {
-                return "15.0";
-            }
-        }
+        internal static string CurrentProductVersion => "16.0";
```
Comment (cdmihai) [inline src/Shared/Constants.cs]: Would it be better if this got read from the dll informational version major part?

**Comment by rainersigwald** [comment]
@dotnet-bot test RHEL7.2 Build for CoreCLR please

**Comment by rainersigwald** [comment]
Superseded by #3779.

**Pull request closed without merging**